common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -246,9 +246,24 @@ class ModuleI18nService(object):
     i18n service.
 
     """
+    def __init__(self):
+        self._xBlock_i18n_info = {}
+
     def __getattr__(self, name):
         return getattr(django.utils.translation, name)
 
+    @property
+    def xBlock_i18n_info(self):
+        """
+        :return _xBlock_i18n_info dictionary
+        xBlock_i18n_info = {
+            'root': xBlock root path,
+            'domain': xBlock domain name,
+            'locale': xBlock locale info
+        }
+        """
+        return self._xBlock_i18n_info
```

Comment:
Why are we still pursuing a design that asks every XBlock to provide information that the runtime can figure out?